### PR TITLE
Improvement/task 1729

### PR DESCRIPTION
**Pull request by albinscania**
**Describe pull-request**  
3 new components for tabs

**Solving issue**  
Fixes: [AB#1729](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/1729) [AB#1836](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/1836) [AB#1837](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/1837)

### CODE REVIEW

**Comment by areblk** [comment]
# Navigation tabs
What I've tested:
- Navigation tab states (default, selected/active, focused, hovered, disabled) styles compared to Figma/UI kit. **Some fixes necessary, see below**
- What happens when tabs combined are wider than viewport? **An arrow appears so user can go to next or previous tab**
- Is there a sticky prop? How does that work? Other props available? **No props/args available**
- Any apparent issues with z-index? **I added a sdds-table and a sdds-dropdown to Storybook without apparent issues. More integration testing could be done in demo pages.**
- Are tabs possible to focus and select (`{Tab}` and `{Enter}`) only using a keyboard? **Yes, including 'disabled' but perhaps it shouldn't. Not sure though, so I'm not suggesting any changes**
- Does it match Usage and Style documentation pages? **No documentation pages available, only Figma UI kit**

## Styling to be fixed
Figma source of truth that I'm comparing against looks like this
image

1. Default state should have color blue-900 (now rgba(0, 21, 51, 0.6))
2. Selected tab bottom border is 2px, but 1px is hidden by parent element bottom border 
image
4. Focused state border looks a bit off. Outline should be on Tab name, not the entire "tab", and not cover Tab name as it does now.
image
5. Focused color should be grey-958 (same as for :selected)
6. Disabled tab color looks OK, but in Figma it's grey-900 with an opacity of 38%. Not sure what variable that is.

## Notes
- I have not tested Inline tabs yet. Review is coming!

**Comment by areblk** [comment]
# Inline tabs
What I've tested:
- Inline tab states (default, selected/active, focused, hovered, disabled) styles compared to Figma/UI kit. **Some fixes necessary, see below**
- What happens when tabs combined are wider than viewport? **An arrow appears so user can go to next or previous tab**
- Any apparent issues with z-index? **I added a sdds-table and a sdds-dropdown to Storybook without some issues. table issues are probably due to table itself, but dropdown issue sould have with this component to do. Even more integration testing could be done in demo pages.**
- Are tabs possible to focus and select ({Tab} and {Enter}) only using a keyboard? **Yes, except 'disabled' which is different from Navigation tabs where disabled tab is focusable. Not sure what's expected, so I'm not suggesting any changes.**
- Does it match Usage and Style documentation pages? **No documentation pages available, only Figma UI kit.**

Figma source of truth that I'm comparing against looks like this
image

## Styling to be fixed
6. Default colour is rgba(0, 21, 51, 0.6) but should be blue-900 with 0.6 opacity
7. Selected should have no right or left border
8. Focused style seem to be missing, should have colour grey-958 with border 1px blue-400, background: grey-100
image
9. Disabled colour is rgba(0, 21, 51, 0.38) but should be blue-900 with 0.16 opacity

## Functionality
10. When placing a dropdown within a tab's `<div>`, only content that fit within that div is shown. It seems too me like an unwanted behavior
image


*Coming up with inline tabs fullbleed review*

**Comment by areblk** [comment]
# Inline Tabs Fullbleed
What I've tested:
- Inline tab states (default, selected/active, focused, hovered, disabled) styles compared to Figma/UI kit. **Some fixes necessary, see below**
- What happens when tabs combined are wider than viewport? **An arrow appears so user can go to next or previous tab**
- Any apparent issues with z-index? **Added dropdown to storybook for quick check and it works as expected, i.e. does not hide behind the tabs**
- Are tabs possible to focus and select ({Tab} and {Enter}) only using a keyboard? **Focusable, including 'disabled'. Can't know if they're able to select because of '#' links.**
- Does it match Usage and Style documentation pages? **No documentation pages available, only Figma UI kit and prototype.**

Figma source of truth that I'm comparing against looks like this (I made my own using the UI kit and setting states). Please note that I've placed tabs on a containerbox on-white.
image

## Styling to be fixed
11. Default colour is rgba(0, 21, 51, 0.6) but should be blue-900 with 0.6 opacity **//Probably same as Inline Tabs. Colour is correct, but I guess it's better to use the variables//**
12. Selected style seems to be missing, colour should be grey-958, bottom border 2px, blue-400
13. Disabled colour is but rgba(0, 21, 51, 0.38) but should be blue-900, opacity 0.38 **//colour is correct, but I guess it's better to use the variables//**
14. Hover bottom border is grey-300 but should be grey-400 (according to Figma UI kit)
15. Hover bottom border should be 2px, but 1px is hidden by  .sdds-inline-tabs-fullbleed border bottom
16. Focused colour is rgba(0, 21, 51, 0.6) but should be grey-958 **//colour is correct, but I guess it's better to use the variables//**
17. Focused state border looks a bit off. Outline should be on Tab name, not the entire "tab", and not cover Tab name as it does now. **//Same as for Inline Tabs//**

## Notes
18. When looking at the prototype and Figma UI kit, I get the feeling that Inline Tabs in fullbleed should have the same background colour as it's parent element so that it can be placed on containerboxes that are white or grey-50 and "melt into them". In prototype, parent element is a containerbox with a headline. My suggestion is to make background adaptable to it's parent, but I think you know the best technical solution for this. I've created a new story to visualise this use case that we can use for testing, `/story/component-tabs-inline-tabs-full-bleed-test--basic`. 

Expected result (sorry about any incorrent padding, didn't know how it's supposed to be)
image

Actual result
image

**Comment by areblk** [comment]
# Final notes
I've created two new stories for increased testability. Please replace original stories with them if you think they're useful.
## 'Inline Tabs (full bleed)' Test
Separate file where story has added controls for headline text, content and number of tabs (minimum 2, maximum 6)
## 'Inline Tabs Test')
Separate file where story has added controls for using sdds-container (will add padding to element), width (i.e. number of cols) and number of tabs (minimum 2, maximum 6)

**Review by areblk — APPROVED** [review]
OK!

File: components/src/components/inline-tabs-fullbleed/inline-tabs-fullbleed.scss [hunk 1]
```diff
@@ -0,0 +1,197 @@
+@import "@scania/typography/dist/scss/mixins";
+@import "@scania/typography/dist/scss/tokens";
+
+.sdds-inline-tabs-fullbleed {
+  display: flex;
+  position: relative;
+
+  //background-color: var(--sdds-grey-50);
```
Comment (theJohnnyMe) [inline components/src/components/inline-tabs-fullbleed/inline-tabs-fullbleed.scss]: If not needed, remove it.

File: components/src/components/inline-tabs/inline-tabs.scss [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+  &-header {
+    position: relative;
+  }
+
+  &-main {
+    background-color: var(--sdds-grey-50);
+    padding: 24px 24px 48px;
+
+    // min-height: 320px;
```
Comment (theJohnnyMe) [inline components/src/components/inline-tabs/inline-tabs.scss:20]: If not needed, please remove it.

File: components/src/components/navigation-tabs/navigation-tabs.scss [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    }
+
+    &::after {
+      content: " ";
+      position: absolute;
+      bottom: 0;
+      right: 0;
+      left: 0;
+      margin-left: auto;
+      width: 0%;
```
Comment (theJohnnyMe) [inline components/src/components/navigation-tabs/navigation-tabs.scss:116]: A minor thing can be just 0 without a unit.

**Review by theJohnnyMe — CHANGES_REQUESTED** [review]

**Comment by theJohnnyMe** [comment]
A minor thing for @AlbinFarstun  - See if it is possible to set prop autoHeight to boolean as anyway it is either on or off. We already talked about it, but just to keep everyone in the loop.

**Review by theJohnnyMe — COMMENTED** [review]
The only things from my side before approval: 
1. make autoHeight a boolean prop to turn ON/OFF height to auto
2. put all versions of inline tabs in the parent "Tab" folder - just for easier finding and to reflect the hierarchy we have in Storybook too.

**Review by theJohnnyMe — APPROVED** [review]
Looks good to me!

**Comment by theJohnnyMe** [comment]
The last change I did before merging was grouping all tabs components under the "Tabs" folder in the code too, so Storybook hierarchy == Code hierarchy. 
Also, rebase with main was done.

**Comment by theJohnnyMe** [comment]
@AlbinFarstun / @albinscania - I leave the branch so you can check changes done. Please delete it when you feel it is good time :)

**Pull request merged**